File: bootstrap/src/test/java/net/dontdrinkandroot/wicket/bootstrap/test/SignInPageTest.java
package net.dontdrinkandroot.wicket.bootstrap.test;

import net.dontdrinkandroot.wicket.bootstrap.page.SignInPage;
import org.apache.wicket.request.mapper.parameter.PageParameters;
import org.junit.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class SignInPageTest extends AbstractWicketTest
{
    @Test
    public void testInstantiation()
    {
        this.tester.startPage(new SignInPage(new PageParameters())
        {
            @Override
            protected boolean isSignedIn()
            {
                return false;
            }

            @Override
            protected boolean signIn(String username, String password)
            {
                return false;
            }
        });
        this.tester.assertRenderedPage(SignInPage.class);
    }
}

File: bootstrap/src/test/kotlin/page/SignInPageTest.kt
package page

import net.dontdrinkandroot.wicket.bootstrap.page.SignInPage
import net.dontdrinkandroot.wicket.bootstrap.test.AbstractWicketTest
import org.apache.wicket.request.mapper.parameter.PageParameters
import org.junit.jupiter.api.Test

class SignInPageTest : AbstractWicketTest() {

    @Test
    fun testInstantiation() {
        tester.startPage(object : SignInPage(PageParameters()) {
            override val signedIn: Boolean
                protected get() = false

            override fun signIn(username: String, password: String): Boolean {
                return false
            }
        })
        tester.assertRenderedPage(SignInPage::class.java)
    }
}